CleverRaven/Cataclysm-DDA
# C4 teleports into pocket when activated outside the pocket #64308

#### Summary
Bugfixes "#64308"


#### Purpose of change



C-4 teleports into Pocket after activating off of floor, 

Steps to Repro:


> ### Steps to reproduce
> 1. Spawn C-4
> 2. Drop it
> 3. `a`ctivate it when it's on the ground
> 4. The game allow you to activate it
> 5. After activation the item teleport into your pockets
> 
> 
> ### Screenshots
> ![GIF 16-03-2023 17-57-56](https://user-images.githubusercontent.com/67688115/225707417-df67b111-a6d6-43ea-a8bc-52f09af6c401.gif)
> 


#### Describe the solution



Changed Flags for Json Objects C-4 

![image](https://user-images.githubusercontent.com/8581536/225820173-63c4d166-f79c-40b9-b142-22aa90dcc21a.png)

The Allows Remote Use/ Radio Modable/ Radio Invoke Proc/ Bomb tags have been added to C4
- This makes it behave similar to a "Gas Drum' Bomb; AND allows for remote control detonation usage

#### Describe alternatives you've considered



I was considering making changes on the C++ side, but when I started adding flags, I noticed that the behavior of C4 was still respected, so I didn't see a need of changing anything in the underlying C++ code located here:


![image](https://user-images.githubusercontent.com/8581536/225806606-afe62480-3672-4953-9285-06d2cef8dc01.png)

#### Testing



I was able to confirm with C4 that I was able to place them and activate without them warping to your pocket
- To Test Allow_Remote_Use:  I performed the following test 
https://user-images.githubusercontent.com/8581536/225964275-9c285bdb-69b6-46eb-bb11-5fe3b9dfcafa.mp4
- To Test Radio Invoke and Radio modable 
  - I spawned in a RadioMod and attached it to a RC Controller  and another RadioMod to Attach the the C4 (Signal Red)
  - Push the red controller button, and you will be prompted to set the c4 timer and it will explode 
- To test BOMB flag 
  - All I did was just make sure the item was throwable; these items are heavy so they can only be thrown 1-2 squares; but if you want to toss the items you can now. 

#### Additional context
